### PR TITLE
Add spawn syscalls for ckb2023

### PR DESCRIPTION
**Pull request by mohanson**
Tests: https://github.com/nervosnetwork/ckb/blob/ckb2023/script/src/verify/tests/ckb_latest/features_since_v2023.rs

Note: The modification of `ckb_utils.h` comes from the command `make fmt`

### CODE REVIEW

File: ckb_syscalls.h [hunk 1]
```diff
@@ -377,6 +377,47 @@ int ckb_exec_cell(const uint8_t* code_hash, uint8_t hash_type, uint32_t offset,
                  argv);
 }
 
+typedef struct {
```
Comment (xxuejie) [inline ckb_syscalls.h]: What about exposing this data structure directly? In general having too many arguments in C(I'm counting 9 or 10 below) is not a good thing.

Comment (mohanson) [reply ckb_syscalls.h]: I redesigned the API, now it takes fewer parameters

File: ckb_syscall_apis.h [hunk 2]
```diff
@@ -42,4 +42,18 @@ int ckb_dlopen2(const uint8_t* dep_cell_hash, uint8_t hash_type,
                 size_t* consumed_size);
 void* ckb_dlsym(void* handle, const char* symbol);
 
+typedef struct {
+  uint64_t memory_limit;
+  int8_t* exit_code;
+  uint8_t* content;
+  uint64_t* content_length;
+} spawn_args;
```
Comment (XuJiandong) [inline ckb_syscall_apis.h:50]: Suggestion:
```
typedef struct spawn_args_t {
  uint64_t memory_limit;
  int8_t exit_code;
  uint8_t* content;
  uint64_t content_length;
} spawn_args_t;
```
1. use `spawn_args_t`
2. `spawn_args_t` is passed as pointer. The `exit_code` can receive values now.
3. Same for `content_length`

File: ckb_syscalls.h [hunk 3]
```diff
@@ -377,6 +377,31 @@ int ckb_exec_cell(const uint8_t* code_hash, uint8_t hash_type, uint32_t offset,
...
+int ckb_spawn_cell(const uint8_t* code_hash, uint8_t hash_type, uint32_t offset,
+                   uint32_t length, int argc, const char* argv[],
+                   spawn_args* spgs) {
+  size_t index = SIZE_MAX;
+  int ret = ckb_look_for_dep_with_hash2(code_hash, hash_type, &index);
+  if (ret != CKB_SUCCESS) {
+    return ret;
+  }
+  size_t bounds = ((size_t)offset << 32) | length;
+  return ckb_spawn(index, CKB_SOURCE_CELL_DEP, bounds, argc, argv, spgs);
```
Comment (XuJiandong) [inline ckb_syscalls.h:394]: Do some checking on `spawn_args_t.memory_limit` and `spawn_args_t.content_length`.

**Review by XuJiandong — CHANGES_REQUESTED** [review]

File: ckb_syscall_apis.h [hunk 4]
```diff
@@ -42,4 +42,18 @@ int ckb_dlopen2(const uint8_t* dep_cell_hash, uint8_t hash_type,
                 size_t* consumed_size);
 void* ckb_dlsym(void* handle, const char* symbol);
 
+typedef struct {
```
Comment (XuJiandong) [inline ckb_syscall_apis.h]: Suggest name: spawn_args_t

**Review by XuJiandong — CHANGES_REQUESTED** [review]

**Review by XuJiandong — APPROVED** [review]

**Pull request merged**